### DISCUSSION TITLE
STL Default "Spinning" Viewing Behavior is visually distracting

### DISCUSSION THREAD

**Discussion by iitsalex** [community/community#62725] [Accessibility] [answered]
### Select Topic Area

Product Feedback

### Body

The stock STL viewer that is utilized for viewing models on github works well for the most part. But it has one *crucial* annoyance that makes it borderline impossible to use for anyone who has visual impairments; the viewer will automatically spin the model clockwise after the model has been clicked on / moved by the user. 

1. Click/drag on model
2. Let go of left click and stop dragging/clicking on the model
3. Model will begin to spin by itself after a few moments.

This is beyond annoying for anyone in the 3D printing community or any other community that needs to look at STLs through Github.

**Comment by cehfisher** [comment]
Hi @iitsalex - thank you for providing this feedback! We wanted to follow up on this reported bug. Is this still an issue you can replicate? If it is resolved, could you please close out this issue? If it is not resolved, could you supply more detailed instructions to reproduce your issue and include screenshots and/or URLs, if applicable. Thank you! (cc @queenofcorgis)

**Reply by iitsalex** [reply]
Definitely still replicable, thanks for reaching back out!

Here is the STL I have chosen to demonstrate: https://github.com/VoronDesign/Voron-Trident/blob/main/STLs/Z_Assembly/z_bed_rear.stl

And here is a reproduction of the issue with the following steps:
1. Open a preview of an stl file, such as [this](https://github.com/VoronDesign/Voron-Trident/blob/main/STLs/Z_Assembly/z_bed_rear.stl)
2. Click anywhere on the preview and drag to rotate the object
3. Release the drag and do not click on the preview again
4. The preview will become spinning on it's own
5. Click again on the preview and it will stop momentarily, before spinning again.
![chrome_fv3eBHbJyq-ezgif com-optimize](https://github.com/community/community/assets/20635452/93885ad9-7c58-4a59-91a8-37fd1dfc740c)

Ideally this feature can be toggled OR just flat out removed. Someone out there may like the spinning "feature" but many of us want the option to remove it.

**Reply by cehfisher** [reply]
Thank you @iitsalex - we will file this with our Accessibility team so that it can be reviewed and will follow-up here with any additional questions or updates. Thanks again!

**Comment by Cellivar** [comment]
Hi @cehfisher, checking in to see if there has been any progress on implementing an accessbility feature to disable or, at the very least, stop the rotation in the STL viewer.

Upon inspecting the viewer javascript it's clear this is intentional behavior. At this URL: https://viewscreen.githubusercontent.com/view/solid?browser=chrome&bypass_fastly=true&color_mode=dark&commit=1402968312766076d115c18b963b86d19861f4a6&device=unknown_device&docs_host=https%3A%2F%2Fdocs.github.com&enc_url=68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f656c65636c61622d7270692f37494e4348345f48444d495f31323830783430302f313430323936383331323736363037366431313563313862393633623836643139383631663461362f37696e6368345f31323830783430305f436173655f5479706533425f466f725f5250492e53544c&logged_in=true&nwo=eleclab-rpi%2F7INCH4_HDMI_1280x400&path=7inch4_1280x400_Case_Type3B_For_RPI.STL&platform=windows&repository_id=471020882&repository_type=Repository&version=126#999fc3ef-933a-46f7-856f-582c2ebe69ee

we can inspect the source code with the Chrome debugger. This leads us to `attract.ts` file:

```
// This mode is responsible for spinning the model if there is no
// user interaction, and not spinning the model when there is any
class AttractMode extends ControlMode {
```

We then find the function `activateAttractMode` in controls.ts that when the interface is idle it will enable AttractMode and begin spinning.

Returning to a browser tab to have a rapidly spinning complex scene is disorienting and annoying. I would also appreciate a way to make it stop.

**Reply by cehfisher** [reply]
Sorry @Cellivar - I do not have any updates to share at this time. I have added your comment to the internal issue tracking this topic though. Thank you for your continued patience as the teams work on this!

**Comment by N-Masi** [comment]
Just wanted to follow up on this, a toggle on this behavior would be great! @cehfisher

**Comment by cehfisher** [comment, marked as answer]
Awesome news @iitsalex – the auto-rotation issue you reported has been resolved! On top of that, the accessibility and service team engineers have added a toggle to enable/disable the rotation, based on the request from @N-Masi 🎉 

Please give it a try on your end and share your feedback with the team. Thanks!

**Reply by iitsalex** [reply]
Fantastic! This is amazing news, especially for those with accessibility issues. Thank you!
